File: spring-mvc/src/main/java/software/darkmatter/school/blog/domain/code/rest/CodeApiClient.java
package software.darkmatter.school.blog.domain.code.rest;

import software.darkmatter.school.blog.domain.code.rest.dto.ApiResult;
import software.darkmatter.school.blog.domain.code.rest.dto.CodeCheckRequest;
import software.darkmatter.school.blog.domain.code.rest.dto.CodeCreateRequest;

public interface CodeApiClient {

    ApiResult createCode(CodeCreateRequest codeCreateRequest);

    ApiResult checkCode(CodeCheckRequest codeCheckRequest);
}

File: kotlin-spring-webflux/src/main/kotlin/software/darkmatter/school/blog/domain/code/rest/CodeApiClient.kt
package software.darkmatter.school.blog.domain.code.rest

import software.darkmatter.school.blog.domain.code.rest.dto.ApiResult
import software.darkmatter.school.blog.domain.code.rest.dto.CodeCheckRequest
import software.darkmatter.school.blog.domain.code.rest.dto.CodeCreateRequest

interface CodeApiClient {

    suspend fun createCode(request: CodeCreateRequest): ApiResult

    suspend fun checkCode(request: CodeCheckRequest): ApiResult
}
